{"cells":[{"metadata":{},"cell_type":"markdown","source":"# Basic info about this Competition (Directly copied from the page)\n**What is the Mechanism of Action (MoA) of a drug? And why is it important?**\n\nIn the past, scientists derived drugs from natural products or were inspired by traditional remedies. Very common drugs, such as paracetamol, known in the US as acetaminophen, were put into clinical use decades before the biological mechanisms driving their pharmacological activities were understood. Today, with the advent of more powerful technologies, drug discovery has changed from the serendipitous approaches of the past to a more targeted model based on an understanding of the underlying biological mechanism of a disease. In this new framework, scientists seek to identify a protein target associated with a disease and develop a molecule that can modulate that protein target. As a shorthand to describe the biological activity of a given molecule, scientists assign a label referred to as mechanism-of-action or MoA for short.\n\n**How do we determine the MoAs of a new drug?**\n\nOne approach is to treat a sample of human cells with the drug and then analyze the cellular responses with algorithms that search for similarity to known patterns in large genomic databases, such as libraries of gene expression or cell viability patterns of drugs with known MoAs.\n\nIn this competition, you will have access to a unique dataset that combines gene expression and cell viability data. The data is based on a new technology that measures simultaneously (within the same samples) human cellsâ€™ responses to drugs in a pool of 100 different cell types (thus solving the problem of identifying ex-ante, which cell types are better suited for a given drug). In addition, you will have access to MoA annotations for more than 5,000 drugs in this dataset.\n\nAs is customary, the dataset has been split into testing and training subsets. Hence, your task is to use the training dataset to develop an algorithm that automatically labels each case in the test set as one or more MoA classes. Note that since drugs can have multiple MoA annotations, the task is formally a multi-label classification problem."},{"metadata":{"trusted":true},"cell_type":"code","source":"import plotly \nplotly.offline.init_notebook_mode (connected = True)","execution_count":null,"outputs":[]},{"metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true},"cell_type":"code","source":"# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nimport plotly.express as px\nimport seaborn as sns \nimport matplotlib.pyplot as plt \n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","execution_count":null,"outputs":[]},{"metadata":{"_uuid":"d629ff2d2480ee46fbb7e2d37f6b5fab8052498a","_cell_guid":"79c7e3d0-c299-4dcb-8224-4455121ee9b0","trusted":true},"cell_type":"code","source":"path = '../input/lish-moa'\nos.listdir(path) ","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"test_features = pd.read_csv('/kaggle/input/lish-moa/test_features.csv')\ntrain_features = pd.read_csv('/kaggle/input/lish-moa/train_features.csv')\ntrain_targets_scored = pd.read_csv('/kaggle/input/lish-moa/train_targets_scored.csv')\ntrain_targets_nonscored = pd.read_csv('/kaggle/input/lish-moa/train_targets_nonscored.csv')\n","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"# let us explore the files "},{"metadata":{},"cell_type":"markdown","source":"### Features for the training set. \n- **Features g-** signify gene expression data, \n- **c-** signify cell viability data. \n- **cp_type -** indicates samples treated with a compound (cp_vehicle) or with a control perturbation (ctrl_vehicle); control perturbations have no MoAs; \n- **cp_time and cp_dose:** indicate treatment duration (24, 48, 72 hours) and dose (high or low)."},{"metadata":{"trusted":true},"cell_type":"code","source":"train_features.head()","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"### train_targets_scored.csv - The binary MoA targets that are scored."},{"metadata":{"trusted":true},"cell_type":"code","source":"train_targets_scored.head()","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"### train_targets_nonscored.csv - Additional (optional) binary MoA responses for the training data. These are not predicted nor scored."},{"metadata":{"trusted":true},"cell_type":"code","source":"train_targets_nonscored.head()","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"# information about the all the csv files  "},{"metadata":{"trusted":true},"cell_type":"code","source":"print('______________train_features________________')\ntrain_features.info()\nprint('______________________________')\nprint('______________test_features________________')\ntest_features.info()\nprint('______________________________')\nprint('_____________train_targets_scored_________________')\ntrain_targets_scored.info()\nprint('______________________________')\nprint('________________train_targets_nonscored______________')\ntrain_targets_nonscored.info()","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"# EDA of training and testing set: \n"},{"metadata":{"trusted":true},"cell_type":"code","source":"\nprint(\"Shape of the training set: \", train_features.shape)\nprint('unique ids: ', len(train_features.sig_id.unique()))\n\nprint(\"Shape of the training set: \", test_features.shape)\nprint('unique ids: ', len(test_features.sig_id.unique()))","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"### for training file:\n- There are 23814 rows and 876 colums in the csv file. \n- there are 23814 unique ids.\n\n### for testing file: \n- There are 3982 rows and 876 colums in the csv file. \n- there are 3982 unique ids."},{"metadata":{"trusted":true},"cell_type":"code","source":"import seaborn as sns \nfig,ax= plt.subplots(1,2)\nsns.countplot(train_features['cp_time'], ax = ax[0]).set_title('For Training Set')\nsns.countplot(test_features['cp_time'], ax = ax[1]).set_title('For Testing Set')\nplt.tight_layout()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"import seaborn as sns \nfig,ax= plt.subplots(1,2)\nsns.countplot(train_features['cp_type'], ax = ax[0]).set_title('For Training Set')\nsns.countplot(test_features['cp_type'], ax = ax[1]).set_title('For Testing Set')\nplt.tight_layout()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"import seaborn as sns \nfig,ax= plt.subplots(1,2)\nsns.countplot(train_features['cp_dose'], ax = ax[0]).set_title('For Training Set')\nsns.countplot(test_features['cp_dose'], ax = ax[1]).set_title('For Testing Set')\nplt.tight_layout()","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"## Let us use Pandas profiler for more exploration "},{"metadata":{"trusted":true},"cell_type":"code","source":"x = train_features.drop(['sig_id'], axis=1)\ncorr = x.corr()\ncorr.style.background_gradient(cmap='coolwarm')","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"# Let us explore the Target of training set "},{"metadata":{},"cell_type":"markdown","source":"# Let us count the positive cases "},{"metadata":{"trusted":true},"cell_type":"code","source":"# drop the first column ('sig_id'), and \ndf = train_targets_scored.drop(['sig_id'], axis=1).sum(axis=0).sort_values(ascending=False).reset_index()\n","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"# positive cases "},{"metadata":{"trusted":true},"cell_type":"code","source":"df.columns = ['column', 'nonzero_records']\ndf","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# plot the bar \n\nfig = px.bar(\n    df.head(50), \n    x='nonzero_records', \n    y='column', \n    orientation='h', \n    title='Columns with the positive samples (Only top 50)', \n    height=1000, \n    width=800\n)\nfig.show()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# drop the first column ('sig_id') and count the 0s in \ndf1 = train_targets_scored.drop(['sig_id'], axis=1).sum(axis=0).sort_values(ascending=False).reset_index()\ndf1.columns = ['column', '% nonzero_records']\ndf1['% nonzero_records'] = (df1['% nonzero_records']/len(train_targets_scored))*100\n# plot the bar \n\nfig = px.bar(\n    df1.head(50), \n    x='% nonzero_records', \n    y='column', \n    orientation='h', \n    title='Columns with the % positive samples (Only top 50) ', \n    height=1000, \n    width=800\n)\nfig.show()\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}